### DISCUSSION TITLE
Specify date-based folder structure for daily notes - how to?

### DISCUSSION THREAD

**Discussion by arvdv** [702573N/Obsidian-Tasks-Calendar#93] [Q&A] [answered]
Hi there,

I make use of a specific daily note format to ensure that they are grouped into years and months in order based on the month. What this looks like is "YYYY/M - MMMM/YYYY-MM-DD", which translates to "1. Timestamps/Daily notes/2024/6 - June/2024-06-21" in my repository if we take today's date as an example.

I've been very happy with how this calendar setup works, but am finding it difficult set either the "dailyNoteFolder" or the "dailyNoteFormat" to support this kind of structure. At the moment, the calendar is unable to find any of my daily notes unless I specify the month using "dailyNoteFolder: "1. Timestamps/Daily notes/2024/6 - June"". Is there a way for me to use the periodic notes' support for dynamically naming folders to pass to either dailyNoteFolder or dailyNoteFormat? I know the readme points out that we can only use a limited set of characters, which crucially don't include forward slashes.

**Comment by bradkoerner** [comment, marked as answer]
I had the same need. It seems like the way to accomplish this with moment.js is to use `[]` to escape text. Unfortunately the code here doesn't fully support it.

The dev seems to be focusing more on Obsidian-Tasks-Timeline, so I forked this and did it myself: [bradkoerner/Obsidian-Tasks-Calendar](https://github.com/bradkoerner/Obsidian-Tasks-Calendar)

Just surround your folder slashes with `[]` and you should get what you're looking for.
```
dailyNoteFolder: "1. Timestamps/Daily notes",
dailyNoteFormat: "YYYY[/]M - MMMM[/]YYYY-MM-DD
```

I also forked the Obsidian-Tasks-Timeline repo and added the same support there: [bradkoerner/Obsidian-Tasks-Timeline](https://github.com/bradkoerner/Obsidian-Tasks-Timeline)

**Reply by arvdv** [reply]
Awesome, thanks - your forked repo and the bracketed slash you suggested did the trick!

**Comment by ctbarber** [comment]
If I understand your question correctly, I was able to use the following to dynamically pull in the current year and month to dynamically build the folder structure.

`dailyNoteFolder: "journal/daily/" + dv.date('today').year + "/" + dv.date('today').month,`
